code/core/viewer.cpp

```diff
@@ -80,6 +81,27 @@ void viewer::AddSelectable(QString id)
     mapScene->addItem(selectableItems[id]);
 }
 
+void viewer::HighlightShop(QString id)
+{
+    qDebug() << selectableItems[id];
+    recoloredItem = selectableItems[id];
```

Comment:
If the map contains no item with key "id", the operator[] inserts a default-constructed value into the map with key "id", but i think it is not what we need and better will just do nothing